state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1739,19 +1745,20 @@ pub mod pallet {
 					};
 
 					let total_input_remaining = swap.input_amount + *remaining_input_amount;
-					let amount_to_refund = match Self::take_refund_fee(
-						total_input_remaining,
-						request.input_asset,
-						matches!(output_action, SwapOutputAction::CreditOnChain { .. }),
-					) {
-						Ok(remaining) => remaining,
-						Err(e) => {
-							log_or_panic!(
+					let FeeTaken { remaining_amount: amount_to_refund, fee: refund_fee } =
+						match Self::take_refund_fee(
+							total_input_remaining,
+							request.input_asset,
+							matches!(output_action, SwapOutputAction::CreditOnChain { .. }),
+						) {
+							Ok(fee_taken) => fee_taken,
+							Err(e) => {
+								log_or_panic!(
 								"Failed to calculate refund fee for swap request {swap_request_id}: {e:?}"
 							);
-							total_input_remaining
-						},
-					};
+								FeeTaken { remaining_amount: total_input_remaining, fee: 0 }
+							},
+						};
 
 					if amount_to_refund > 0 {
```

Comment:
I think want to emit `RefundEgressIgnored` when `amount_to_refund <= 0`. (see this [discussion](https://discord.com/channels/775961728608895008/1044538771967840276/1372589802926641277) for context). Could be done in a separate PR though.